### PR TITLE
Issue 135 - Feature to limit number of builds to keep.

### PR DESCRIPTION
**Pull request by suratdas**
Addressed reported issue https://github.com/Visual-Regression-Tracker/Visual-Regression-Tracker/issues/135

### CODE REVIEW

File: src/components/ProjectForm/ProjectForm.tsx [hunk 1]
```diff
@@ -71,6 +71,26 @@ export const ProjectForm: React.FunctionComponent = () => {
           })
         }
       />
+      <TextValidator
+          name="buildCount"
+          validators={["minNumber:1"]}
+          errorMessages={["Enter greater than 0"]}
```
Comment (pashidlos) [inline src/components/ProjectForm/ProjectForm.tsx:77]:
```suggestion
          errorMessages={["Enter greater than 1"]}
```

File: src/contexts/build.context.tsx [hunk 2]
```diff
@@ -197,6 +197,10 @@ async function deleteBuild(dispatch: Dispatch, id: string) {
   });
 }
 
+async function clearBuild(dispatch: Dispatch, id: string) {
```
Comment (pashidlos) [inline src/contexts/build.context.tsx:200]: let's instead directly dispatch actions, if it doesn't contain any api calls or additional logic

File: src/contexts/socket.context.tsx [hunk 3]
```diff
@@ -79,6 +81,13 @@ function SocketProvider({ children }: SocketProviderProps) {
         );
       });
 
+      state.socket.on("build_deleted", function (build: Build) {
+        clearBuild(buildDispatch, build.id);
```
Comment (pashidlos) [inline src/contexts/socket.context.tsx:85]:
```suggestion
        buildDispatch({ type: "delete", payload: build.id });
```

File: src/contexts/socket.context.tsx [hunk 4]
```diff
@@ -79,6 +81,13 @@ function SocketProvider({ children }: SocketProviderProps) {
...
+        if (selectedBuild && selectedBuild.id === build.id) {
```
Comment (pashidlos) [inline src/contexts/socket.context.tsx:86]: could you try reusing logic from `BuildList` not to duplicate it here?
```js
if (indexOfBuildDeleted === indexOfSelectedBuild) {
                  if (buildList.length > 1) {
                    if (indexOfBuildDeleted === 0) {
                      selectBuild(buildDispatch, buildList[1].id);
                    } else {
                      selectBuild(
                        buildDispatch,
                        buildList[indexOfBuildDeleted - 1].id
                      );
                    }
                  } else {
                    selectBuild(buildDispatch, null);
                  }
                }
```

**Review by pashidlos — CHANGES_REQUESTED** [review]

File: src/contexts/socket.context.tsx (hunk 3 above)
Comment (pashidlos) [reply src/contexts/socket.context.tsx:85]: let's instead directly dispatch actions, if it doesn't contain any api calls or additional logic

File: src/contexts/socket.context.tsx (hunk 4 above)
Comment (suratdas) [reply src/contexts/socket.context.tsx]: Moved the select build after deleting logic to the context now.

**Review by pashidlos — APPROVED** [review]
LGTM!

**Pull request merged**